apps/web/src/components/dashboard/csv-export-button.tsx

```diff
@@ -0,0 +1,106 @@
+"use client";
+
+import {
+    AlertDialog,
+    AlertDialogAction,
+    AlertDialogCancel,
+    AlertDialogContent,
+    AlertDialogDescription,
+    AlertDialogFooter,
+    AlertDialogHeader,
+    AlertDialogTitle,
+    AlertDialogTrigger,
+    Button,
+} from "@energyleaf/ui";
+import { DateRangePicker } from "@energyleaf/ui/components/utils";
+import { DownloadIcon } from "lucide-react";
+import { useMemo, useState } from "react";
+import type { DateRange } from "react-day-picker";
+import { toast } from "sonner";
+
+interface Props {
+    userId: string;
+    userHash: string;
+    endpoint: string;
+}
+
+export default function CSVExportButton({ userId, userHash, endpoint }: Props) {
+    const [open, setOpen] = useState(false);
+    const [range, setRange] = useState<DateRange>({
+        from: new Date(0),
+        to: new Date(),
+    });
+    const [loading, setLoading] = useState(false);
+
+    function startDownload() {
+        if (loading) return;
+        setLoading(true);
+        toast.promise(
+            async () => {
+                const res = await fetch(endpoint, {
+                    method: "POST",
+                    body: JSON.stringify({
+                        userId,
+                        userHash,
+                        start: range.from?.toISOString(),
+                        end: range.to?.toISOString(),
+                    }),
+                });
+
+                if (!res.ok) {
+                    throw new Error("Unable to contact endpoint");
+                }
+
+                if (res.headers.get("Content-Type") === "application/json") {
```

Comment:
Warum nicht genereller?
```suggestion
                if (res.headers.get("Content-Type") !== "text/csv") {
```

Reply:
Auch möglich.